libs/contrib/Data/List/Lazy.idr

```diff
@@ -98,6 +98,12 @@ Foldable LazyList where
   null []     = True
   null (_::_) = False
 
+  foldMap f xs = go xs neutral
+    where
+      go : LazyList a -> m -> m
+      go [] z = z
+      go (x :: xs) z = go xs (z <+> f x)
+
```

Comment:
`foldMap` for lazy lists mustn't be tail-recursive as it forces a lazy list to be always fully evaluated. This leads to too eager behaviour in a case when you map values of a lazy list to some lazy type (e.g., when `m` is `Lazy Bool`) and them are folding over its monoid.